src/main/java/de/eposcat/master/approachImpl/EAV_DatabaseAdapter.java

```diff
@@ -6,150 +6,170 @@
+    private static final String FIND_PAGE_BY_ATTRIBUTE_QUERY = "SELECT eav_values.ent_id FROM attributes Inner JOIN eav_values ON eav_values.att_id = attributes.id WHERE attributes.name = ?";
+
+    private final Connection conn;
+    private final static String ENTITY_TABLE =  "entityTable";
+    private final static String ATTRIBUTE_TABLE = "attributeTable";
+
     public EAV_DatabaseAdapter(AbstractConnectionManager connectionManager) {
         this.conn = connectionManager.getConnection();
     }
 
     @Override
-    public Page createPage(String typeName) throws SQLException{
+    public Page createPage(String typeName) throws SQLException {
+        if(typeName == null || typeName.isEmpty()){
+            throw new IllegalArgumentException("Typename has to be a non empty String");
+        }
+
         PreparedStatement st = conn.prepareStatement(CREATE_PAGE_QUERY, Statement.RETURN_GENERATED_KEYS);
         st.setString(1, typeName);
-        
+
         int updatedRows = st.executeUpdate();
-        
-        if(updatedRows > 0 ) {
-            ResultSet keyset = st.getGeneratedKeys();
-            Page page = new Page(getIdFromGeneratedKeys(keyset, ENTITYTABLE), typeName);
-            
-            return page;
-        } 
-        
-        return null;
+
+        if (updatedRows > 0) {
+            //TODO? Cancel Transaction if getting generated keys fails?
+            ResultSet keySet = st.getGeneratedKeys();
+            return new Page(getIdFromGeneratedKeys(keySet, ENTITY_TABLE), typeName);
+        }
+
+        throw new SQLException("No new row has been created in database. This should never happen!");
     }
-    
+
+    @Override
+    public Page createPageWithAttributes(String typename, Map<String, Attribute> attributes) throws SQLException{
+        if(attributes == null){
+            throw new IllegalArgumentException("Argument map must not be null");
+        }
+
+        //Currently implementation in two transactions
+        //We cant get much efficiency out of making an extra sql instruction, since it's always a multi step process
+        Page page = createPage(typename);
+        page.setAttributes(attributes);
+        updatePage(page);
+        return page;
+    }
+
     @Override
     public void updatePage(Page page) throws SQLException{
         try {
             conn.setAutoCommit(false);
-            
-            //Erst schauen ob dirty?
+
+            //Check if dirty?
             PreparedStatement st = conn.prepareStatement(UPDATE_PAGE_QUERY);
             st.setString(1, page.getTypeName());
-            st.setInt(2, page.getId());
+            st.setLong(2, page.getId());
             int affectedRows = st.executeUpdate();
-            
+
             if(affectedRows != 1) {
-                throw new SQLException();
+                throw new BlException("page is not tracked by database, try create pageWithAttributes first");
```

Comment:
The pageId (or some page info) would be useful in the exception message, e.g.
```suggestion
                throw new BlException("Page with id=xyz, type=xyz is not tracked by database, try create pageWithAttributes first");
```